### PR TITLE
Passing table name to sp_cursor call

### PR DESCRIPTION
**Pull request by tkyc**
Fix for https://github.com/microsoft/mssql-jdbc/issues/2067

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 import java.util.UUID;
 
 import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.SQLServerResource;
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java:35]: Unused
```suggestion

```

File: src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java [hunk 2]
```diff
@@ -283,6 +293,53 @@ public void testJdbc41ResultSetMethods() throws SQLException {
         }
     }
 
+    @Test
+    public void testUpdateOnColumnForAssociatedTable() throws SQLException {
+        String query1 = "SELECT t.* FROM " + tableName1 + " k INNER JOIN " + tableName2 + " t ON  t.i = k.i";
+        String query2 = "SELECT * FROM " + tableName2;
+        String data = "NEW EDIT";
+
+        try (SQLServerConnection conn = (SQLServerConnection) PrepUtil.getConnection(connectionString)) {
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java:302]: Unneeded cast
```suggestion
        try (SQLServerConnection conn = PrepUtil.getConnection(connectionString)) {
```

File: src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java [hunk 3]
```diff
@@ -283,6 +293,53 @@ public void testJdbc41ResultSetMethods() throws SQLException {
...
+            }
+        }
+    }
+
+    @Test
+    public void testErrorOnAmbiguousUpdate() throws SQLException {
+        String query1 = "SELECT t.*, k.* FROM " + tableName1 + " k INNER JOIN " + tableName2 + " t ON  t.i = k.i";
+        String data = "NEW EDIT";
+
+        try (SQLServerConnection conn = (SQLServerConnection) PrepUtil.getConnection(connectionString)) {
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java:326]: Unneeded cast
```suggestion
        try (SQLServerConnection conn = PrepUtil.getConnection(connectionString)) {
```

File: src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java [hunk 4]
```diff
@@ -636,11 +693,10 @@ public void testResultSetFetchBufferSqlErrorState() throws Exception {
     @Test
     public void testResultSetClientCursorInitializerSqlErrorState() throws Exception {
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java:694]: No exception is ever thrown
```suggestion
    public void testResultSetClientCursorInitializerSqlErrorState() {
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java [hunk 5]
```diff
@@ -4912,6 +4912,23 @@ final boolean hasUpdatedColumns() {
...
     }
 
+    private String getUpdatedColumnTableName() throws SQLServerException {
+        String columnTableName = "";
+
+        for (Column column : columns) {
+            if (column.hasUpdates() && columnTableName.isEmpty()) {
+                columnTableName = column.getTableName().asEscapedString();
+            } else if (column.hasUpdates() && !columnTableName.isEmpty()
+                    && !columnTableName.equals(column.getTableName().asEscapedString())) {
```
Comment (Jeffery-Wasty) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java:4922]: Is the isEmpty() needed here?
```suggestion
            } else if (column.hasUpdates() && !columnTableName.equals(column.getTableName().asEscapedString())) {
```

Comment (tkyc) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java:4922]: Yes, if it is not empty it means it was already set prior. If it was set prior, then we need to check if it matches the prior name. If it doesn't match then we throw.

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]

**Pull request merged**